### PR TITLE
home-work

### PR DESCRIPTION
**Pull request by ddryzhov**
Replaced the existing BookRepository with JpaRepository, added Liquibase support, changed spring.jpa.hibernate.ddl-auto property value to validate, implemented soft delete concept, added put and delete endpoints

### CODE REVIEW

File: src/main/java/mate/academy/bookstore/model/Book.java [hunk 1]
```diff
@@ -5,15 +5,21 @@
...
 import lombok.ToString;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
 
 @Entity
 @Getter
 @Setter
 @ToString
+@SQLDelete(sql = "UPDATE book SET is_deleted = true WHERE id=?")
+@Where(clause = "is_deleted=false")
```
Comment (okuzan) [inline src/main/java/mate/academy/bookstore/model/Book.java]: where is deperacted, replace with SQLRestriction
uppercase true and false 
```suggestion
@SQLRestriction("is_deleted = FALSE")
```

**Review by okuzan — CHANGES_REQUESTED** [review]
extract /api to project-wide prefix (use application.properties)

File: src/main/java/mate/academy/bookstore/service/BookServiceImpl.java [hunk 2]
```diff
@@ -36,4 +36,24 @@ public BookDto findById(Long id) {
...
         return bookMapper.toDto(book);
     }
+
+    @Override
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
+    }
+
+    @Override
+    public BookDto update(Long id, BookDto bookDto) {
```
Comment (MAMentorFX) [inline src/main/java/mate/academy/bookstore/service/BookServiceImpl.java:48]: @Transactional. You make more than one call to DB

File: src/main/java/mate/academy/bookstore/service/BookServiceImpl.java [hunk 3]
```diff
@@ -36,4 +36,24 @@ public BookDto findById(Long id) {
...
+        Book existingBook = bookRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Can't find book by id " + id));
+
+        existingBook.setTitle(bookDto.getTitle());
+        existingBook.setAuthor(bookDto.getAuthor());
+        existingBook.setDescription(bookDto.getDescription());
+        existingBook.setIsbn(bookDto.getIsbn());
+        existingBook.setCoverImage(bookDto.getCoverImage());
+        existingBook.setPrice(bookDto.getPrice());
```
Comment (MAMentorFX) [inline src/main/java/mate/academy/bookstore/service/BookServiceImpl.java]: use MapStruct

**Review by MAMentorFX — CHANGES_REQUESTED** [review]

**Review by olekskov — APPROVED** [review]

**Pull request merged**